.github/workflows/stackhpc-all-in-one.yml

```diff
@@ -94,7 +110,8 @@ jobs:
       - name: Checkout ${{ inputs.upgrade && 'previous release' || 'current' }} config
         uses: actions/checkout@v4
         with:
-          ref: ${{ inputs.upgrade && env.PREVIOUS_BRANCH || github.ref }}
+          repository: ${{ inputs.repository }}
+          ref: ${{ inputs.upgrade && env.PREVIOUS_BRANCH || inputs.called_from_external && inputs.external_ref_override || github.ref }}
```

Comment:
Just to minimise the number of variables we're adding here, do you think we can rename `external_ref_override` to `github_ref` and default it to `github.ref`, then just say `${{ inputs.upgrade && env.PREVIOUS_BRANCH || github_ref }}`

It then makes this conditional a bit more simple and removes the need for `called_from_external`